### PR TITLE
Add citation as `entry.citation` and implemented citation in `cv_entry.md`

### PR DESCRIPTION
**Pull request by DunklesArchipel**
Adds a citation to the `cv_entry.md` which can be called with `entry.citation` . The latter returns a markdown string.
Also fixes the DOI link.

Part of #31.

#103 has to be merged before.

### CODE REVIEW

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:105]:
```suggestion
        Returns a formatted reference from the entry's bibliography of style:
```

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:103]: Shouldn't this be a `@property`?

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:129]: Do we really want the title in single quotes? Maybe that's customary in your field, so just wondering.

Anyway, bad things will happen here if any of these strings (in particular) title contains anything that has a meaning in markdown, in particular of course `<`.

Should we try to use some library to construct such markdown snippets and hope that such library gets the escaping right?

**Review by saraedum — CHANGES_REQUESTED** [review]

**Comment by saraedum** [comment]
Btw., I am not entirely sure how GitHub works here, but I think if you write `Fix...#123` then merging that PR will close the issue so reworded your summary slightly.

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [reply echemdb/data/cv/entry.py:103]: Sure at the moment, it is indeed a property.
I was wondering if we want to provide a bit more flexibility to the output style. for example to provide an option to enable/disable the title or to print all authors. until then I guess it can be a property.

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [reply echemdb/data/cv/entry.py:129]: Actually, every journal has its own style. To be honest I don't have any preferences. Writing it only in italics would also be fine for me.

Comment (DunklesArchipel) [reply echemdb/data/cv/entry.py:129]: Maybe this could solve the issue? https://pypi.org/project/markdown-strings/

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [reply echemdb/data/cv/entry.py:103]:
```suggestion
    @property
    def citation:
```

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [inline echemdb/data/cv/entry.py:113]:
```suggestion
            >>> entry.citation
```

File: templates/cv_entry.md
Comment (DunklesArchipel) [inline templates/cv_entry.md:15]:
```suggestion
[{{ entry.citation }}](https://doi.org/{{ entry.source.doi }}).
```

File: echemdb/data/cv/entry.py
Comment (saraedum) [reply echemdb/data/cv/entry.py:130]: That package seems to store everything as plain strings, see https://github.com/awesmubarak/markdown_strings/issues/8. It might be good enough for our application but you cannot chain commands with an approach like this.
(I struggled with this a bit at https://github.com/standardese/standardese where I had to store everything as strings as well and eventually settled with parsing all arguments as MarkDown before processing them…)
Also, that project has not seen a release since two years almost.

Comment (saraedum) [reply echemdb/data/cv/entry.py:130]: I wonder if we should put all MarkDown generation into proper templates, i.e., have a `citation.md` template that reads something like:
```md
{{ persons['author'].0.first_names.0 | abbreviate }} {{ persons['author'].0.last_names.0 }} {{ TODO: et al here }}, **{{ fields.volume }}** ({{ fields.year }}) {{ fields.pages }} "*{{ fields.title }}*"
```
That template could then be the string rendering of `.bibliography` which is currently not terribly useful.

Together with #78 that could work. This approach also solves #70 and #42.

Comment (DunklesArchipel) [reply echemdb/data/cv/entry.py:130]: Maybe we should discuss this. Currently, I can't imagine how this works, i.e,,
* will `entry.citation` actually call `citation.md` if it is used not to generate the website but for example in a jupyter project?
* what is the difference in creating for example the title string in `entry.citation` than generating it directly in the markdown template as `{{ fields.title }}`. (should be `entry.fields.title`?)

Comment (saraedum) [reply echemdb/data/cv/entry.py:130]: > will entry.citation actually call citation.md if it is used not to generate the website but for example in a jupyter project?

Not exactly. I guess that `str(entry.bibliograpy)` would render `citation.md` internally. There is no need for `entry.citation` anymore.

> what is the difference in creating for example the title string in entry.citation than generating it directly in the markdown template as {{ fields.title }}. (should be entry.fields.title?)

I am not entirely sure what is your question here. This is very similar, in one case, Python code is running in a `.py` file, in the other case very similar code is running in a template.

Historically, it turned out to be a good idea to use templates to create websites and not to produce strings that encode a website in code. Mostly, because templates are much easier to work with in practice.

The entire `cv_entry.md` could be  a method of `Entry` instead of being a template. In that case, it seems obvious (to me at least) that a template is the correct solution here. In the case of the citation I am less convinced that this has to be a template but since the code you wrote performs a lot of string manipulations, it feels reasonable to use a templating engine instead.

**Comment by DunklesArchipel** [comment]
@saraedum : This PR is now ready for review.

The citation is not yet properly rendered which is an issue with the website configuration (see #83).

Currently, only DOI links are supported, where the link is created from `https://doi.org/` and the DOI provided in the metadata file. This will be a problem with PR #51 where a URL is provided in the DOI entry.
An issue was created in #84.

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [inline echemdb/data/cv/entry.py:123]: Apparently, this does not work properly when the author's name contains latex encoding. (see the briega_martos_2021 ... entries when they are rendered. 
This was fixed when I used the 'pylatexenc' module at this point, which renders this latex string to text. Unless someone has a better idea, I would change this back.

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [inline echemdb/data/cv/entry.py:116]: I tried making a text with md rendering, but that's simply a mess. I hope this is fine.

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [inline echemdb/data/cv/entry.py:110]: I would prefer to add a link to the `pybtex` documentation, but I did not find anything useful in the API documentation.
So probably we leave it to the user to find its way through `pybtex` code.

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [inline echemdb/data/cv/entry.py:130]: I had to set all separators manually, that's why it looks a bit messy. :)

**Comment by DunklesArchipel** [comment]
@saraedum did you decide if you want to implement the citation style as an MD template? We could also open an issue if we want to change this at a later stage.

Otherwise, from my side this could be merged.

**Comment by saraedum** [comment]
> @saraedum did you decide if you want to implement the citation style as an MD template? We could also open an issue if we want to change this at a later stage.
> 
> Otherwise, from my side this could be merged.

Yes, I think we had agreed that we want to do the entire citation style with pybtex from Python code.

**Comment by saraedum** [comment]
We should discuss whether we want to use #81 here. Maybe we can talk about this tomorrow.

File: echemdb/data/cv/entry.py
Comment (saraedum) [reply echemdb/data/cv/entry.py]: Sorry, I forgot that there is no actual bibliography in the YAML.

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:105]:
```suggestion
        Return a formatted reference for the entry's bibliography either as:
```

File: echemdb/data/cv/entry.py
Comment (saraedum) [reply echemdb/data/cv/entry.py:123]: The problem is that abbreviate turns the name into a string and then pybtex does not know anymore that this is LaTeX.

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:125]: The leadingspace of the et al. is wrong I think. The `*` block should surround `et al.` and not ` et al.`.

File: echemdb/data/cv/entry.py
Comment (saraedum) [reply echemdb/data/cv/entry.py:130]: I believe that you are not really using pybtex correctly. Let me propose something else :)

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:139]:
```suggestion
                def names(_, context, role):
```

File: echemdb/data/cv/entry.py
Comment (saraedum) [inline echemdb/data/cv/entry.py:151]:
```suggestion
                names = names(role)
```

File: echemdb/data/cv/entry.py
Comment (DunklesArchipel) [reply echemdb/data/cv/entry.py:147]: I added a TODO and updated a related issue. I do not yet completely understand how to style the citation adequately. 

In any case, I think for now it is important to cite the work presented on the website. Style is less important.

**Review by saraedum — APPROVED** [review]
Ok. I think that this can be merged once the CI is happy.

**Pull request merged**
